velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -209,4 +209,26 @@ struct LastDayFunction {
   }
 };
 
+template <typename T>
+struct DateSubFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Date>& result,
+      const arg_type<Date>& date,
+      const int32_t value) {
+    constexpr int32_t kMin = std::numeric_limits<int32_t>::min();
+    if (value > kMin) {
+      int32_t subValue = 0 - value;
+      result = addToDate(date, DateTimeUnit::kDay, subValue);
+    } else {
+      // If input value is kMin, getting subValue(0-value) lead to overflow.
```

Comment:
Thanks. Perhaps, rephrase for clarity.

```
If input values is kMin,  0 - value overflows. Subtract kMin in 2 steps to avoid overflow: - (- (kMin + 1)), then - 1.
```

I have a question though. if 'date' is some positive integer, adding kMin to it would overflow anyway, no? What happens when we call date_sub on 2023-07-10 and kMin?

Reply:
These comments are more clear. Thanks. 
When your mentioned cases happened, overflow happened. Velox code and Spark show the same results. Details see below logs:
spark-sql> SELECT date_sub('1969-12-31', 2147483647);
-5877641-06-23
spark-sql> SELECT date_sub('1969-12-30', 2147483647);
+5881580-07-11
spark-sql> SELECT date_sub('1969-12-31', -2147483648);
+5881580-07-11
spark-sql> SELECT date_sub('1970-01-01', -2147483648);
-5877641-06-23
spark-sql> SELECT date_sub('2023-07-10', -2147483648);
-5877588-12-29

Reply:
> spark-sql> SELECT date_sub('2023-07-10', -2147483648);
> -5877588-12-29

Here, looks like we wrap around, not overflow. Is this how Spark behaves? Let's document this behavior and add this case to the test for illustration.

Reply:
yes, spark is the same behavior.